src/imap-client.js

```diff
@@ -919,9 +921,15 @@
 
         if (options.answered === true) {
             add.push(ANSWERED_FLAG);
-        } else if (options.answered === true) {
+        } else if (options.answered === false) {
             remove.push(ANSWERED_FLAG);
         }
+        
+        if (options.answered === true) {
```

Comment:
browserbox already does that for you. in terms of what's expensive or not, that's the server's problem. i don't really care about how much work it is for google to manage a mailbox.

[browserbox #delete()](https://github.com/whiteout-io/browserbox/blob/master/src/browserbox.js#L1083-1106)
[browserbox #move()](https://github.com/whiteout-io/browserbox/blob/master/src/browserbox.js#L1195-L1227)